lib/livebook/hubs.ex

```diff
@@ -211,21 +211,54 @@ defmodule Livebook.Hubs do
   @doc """
   Creates a secret for given hub.
   """
-  @spec create_secret(Secret.t()) :: :ok | {:error, list({String.t(), list(String.t())})}
+  @spec create_secret(Secret.t()) :: :ok | {:error, list({atom(), list(String.t())})}
   def create_secret(%Secret{origin: {:hub, id}} = secret) do
     case get_hub(id) do
       {:ok, hub} ->
-        if capability?(hub, [:secrets]) do
-          Provider.create_secret(hub, secret)
-        else
-          {:error, Secrets.add_secret_error(secret, :origin, "is invalid")}
-        end
+        if capability?(hub, [:create_secret]),
+          do: Provider.create_secret(hub, secret),
+          else: {:error, add_invalid_origin_error(secret)}
 
       :error ->
-        {:error, Secrets.add_secret_error(secret, :origin, "is invalid")}
+        {:error, add_invalid_origin_error(secret)}
     end
   end
 
+  @doc """
+  Updates a secret for given hub.
+  """
+  @spec update_secret(Secret.t()) :: :ok | {:error, list({atom(), list(String.t())})}
+  def update_secret(%Secret{origin: {:hub, id}} = secret) do
+    case get_hub(id) do
+      {:ok, hub} ->
+        if capability?(hub, [:update_secret]),
+          do: Provider.update_secret(hub, secret),
+          else: {:error, add_invalid_origin_error(secret)}
+
+      :error ->
+        {:error, add_invalid_origin_error(secret)}
+    end
```

Comment:
```suggestion
    with {:ok, hub} <- get_hub(id),
         true <- capability?(hub, [:update_secret]) do
      Provider.update_secret(hub, secret)
    else
      :error -> {:error, add_invalid_origin_error(secret)}
    end
```

Reply:
such `else` would not handle capability?/2 returning false.

Personally (and so take it with a grain of salt) I'd revert all the way back to what was on the main. I think `if do else end` is preferred over `if do:, else:` because the former allows to do printf-debugging more easily and that's the only way I learned to work. :)

Reply:
Whoops, I meant `_`!

Definitely agreed on `if do else end` :)